scripts/core/loader.yml

```diff
@@ -225,3 +225,29 @@ ordinal: "1"
 params: []
 returns:
     - $X_RESULT_ERROR_OUT_OF_HOST_MEMORY
+--- #--------------------------------------------------------------------------
+type: fptr_typedef
+desc: "Callback function to retrieve output from the logger."
+name: $x_logger_output_callback_t
+return: "void"
+params:
+    - type: const char*
+      name: pLoggerMsg
+      desc: "[in][out] pointer to data to be passed to callback"
+    - type: void*
+      name: pUserData
+      desc: "[in][out] pointer to data to be passed to callback"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Set a callback function for use by the logger to retrieve logging output."
+class: $xLoaderConfig
+loader_only: True
+name: SetLoggerCallback
+decl: static
+params:
+    - type: $x_logger_output_callback_t
+      name: pfnLoggerCallback
+      desc: "[in] Function pointer to callback from the logger."
+    - type: void*
+      name: pUserData
+      desc: "[in][out][optional] pointer to data to be passed to callback"
```

Comment:
The way this is implemented right now is not thread safe with respect to any other API call. This should be stated clearly in the docs.
(Ideally we'd figure out how to make it thread-safe though, but I don't see a simple way other than having a mutex around the logger, and that's just iffy).

Reply:
I've updated the description here to say the creator of the callback function should be responsible for making it thread-safe.